### DISCUSSION TITLE
Update StateProvider's state with function

### DISCUSSION THREAD

**Discussion by masterodi** [rrousselGit/riverpod#1088] [Q&A]
I know using StateNotifier you can update the state by calling functions. 
I was wondering if there is any way to update a StateProvider state by calling functions. StateNotifier seems overkill sometimes, just for some simple value assignments.

For example:
```
class SomeState {
  final int counter;

  const SomeState({this.counter=1});

  SomeState resetCounter() => const SomeState()
  SomeState update(int counter) => SomeState(counter: counter);
}

final someStateProvider = StateProvider<SomeState>((ref) => const SomeState());
```

And on a Button callback, instead of doing
```
onPressed: () => ref.read(someStateProvider.notifier).state = const SomeState()
```
I would do
```
onPressed: ... // somehow call resetCounter() if possible
```

Thanks.

**Comment by rrousselGit** [comment]
This is the goal of StateNotifier/StateNotifierProvider

I'd discourage trying to use StateProvider at large scale. It's lacking the organization that StateNotifierProvider provides

**Comment by masterodi** [comment]
I understand, but I was thinking that, if I want, for example, something very simple like a class to store some filter properties which change based on some inputs. I could create a StateProvider with a FilterClass and just change that filter class for this use case. With StateNotifier you have to create a StateNotifier class just to create some simple setters.
I know that StateNotifier is made with that logic in mind, to change the state through functions.

Also, I don't know if this is the right place to ask you, but, regarding the StateProvider, are ```ref.read(stateProvider.state)``` and ```ref.read(stateProvider.notifier)``` the same thing? Looks like they return the same type. I'm a bit confused.
